src/Symfony/Component/Yaml/Inline.php

```diff
@@ -485,6 +485,14 @@ private static function parseMapping($mapping, $flags, &$i = 0, $references = ar
                 @trigger_error('Omitting the key of a mapping is deprecated and will throw a ParseException in 4.0.', E_USER_DEPRECATED);
             }
 
+            if (!(Yaml::PARSE_KEYS_AS_STRING & $flags)) {
+                $evaluatedKey = self::evaluateScalar($key, $flags, $references);
+
+                if ($evaluatedKey !== $key && !is_string($evaluatedKey) && !is_int($evaluatedKey)) {
```

Comment:
`!is_int($evaluated)` should be removed imo, it is not supported currently so it should trigger a deprecation too.